### PR TITLE
Add "RequireAttestationsBy" into ImageSecurityPolicy for custom attestation check

### PR DESCRIPTION
**Pull request by dragon3**
## WHAT

This pull request adds "RequireAttestationsBy" into ImageSecurityPolicy for custom attestation check.
For the required attestation check, use Binary Authorization API ( [projects.attestors](https://cloud.google.com/binary-authorization/docs/reference/rest/v1beta1/projects.attestors) ) to get attestors with their PGP public keys.

## WHY

Currently Kritis doesn't have a feature to verify attestations like GCP Binary Authorization.
This pull request adds the functionality.

## REF

- https://groups.google.com/forum/#!msg/kritis-users/pzG2zD0pgkc/NCpMfCASAwAJ

### CODE REVIEW

File: pkg/kritis/crd/securitypolicy/securitypolicy.go [hunk 1]
```diff
@@ -194,3 +239,101 @@ func severityWithinThreshold(maxSeverity string, severity string) (bool, error)
...
+
+func fingerprint(key string) (string, error) {
+	s, err := secrets.NewPgpKey("", "", key)
+	if err != nil {
+		return "", err
+	}
+	return s.Fingerprint(), nil
+}
+
+func NewBinauthzAttestorFetcher() (AttestorFetcher, error) {
```
Comment (dtan4) [inline pkg/kritis/crd/securitypolicy/securitypolicy.go]: This function is not used anywhere and duplicated with `NewAttestorFetcher`.

File: pkg/kritis/review/review.go [hunk 2]
```diff
@@ -67,31 +69,32 @@ func (r Reviewer) Review(images []string, isps []v1beta1.ImageSecurityPolicy, po
...
-				return fmt.Errorf("error validating image security policy %v", err)
+				return fmt.Errorf("failed validating image security policy: %v", err)
 			}
 			if len(violations) != 0 {
 				return r.handleViolations(image, pod, violations)
 			}
 			if r.config.IsWebhook {
 				if err := r.addAttestations(image, attestations, isp); err != nil {
-					glog.Errorf("error adding attestations %s", err)
+					glog.Errorf("Failed adding attestations: %s", err)
```
Comment (dtan4) [inline pkg/kritis/review/review.go]: Please use the same case: `failed`

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: pkg/kritis/crd/securitypolicy/securitypolicy.go (hunk 1 above)
Comment (dragon3) [reply pkg/kritis/crd/securitypolicy/securitypolicy.go]: Oops, good catch.
Fixed in d586751

File: pkg/kritis/review/review.go (hunk 2 above)
Comment (dragon3) [reply pkg/kritis/review/review.go]: Fixed in f528ceb

File: pkg/binauthz/binauthz.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package binauthz
```
Comment (dtan4) [inline pkg/binauthz/binauthz.go]: How about put this file in `pkg/kritis` (e.g., `pkg/kritis/binauthz/binauthz.go`) ?

Comment (dragon3) [reply pkg/binauthz/binauthz.go]: OK. I'm going to move it.

Comment (dragon3) [reply pkg/binauthz/binauthz.go]: Fixed in 69bd491

**Review by dtan4 — APPROVED** [review]
LGTM 👍

**Comment by dragon3** [comment]
Thank you for the review.
I'm going to merge this PR ( into `release-201905` ).

**Pull request merged**